packages/nc-gui/components/cell/GeoData.vue

```diff
@@ -0,0 +1,154 @@
+<script lang="ts" setup>
+import type { GeoLocationType } from 'nocodb-sdk'
+import { Modal as AModal, useVModel } from '#imports'
+import { latLongToJoinedString } from '~~/utils/geoDataUtils'
+
+interface Props {
+  modelValue?: string | null
+}
+
+interface Emits {
+  (event: 'update:modelValue', model: GeoLocationType): void
+}
+
+const props = defineProps<Props>()
+
+const emits = defineEmits<Emits>()
+
+const vModel = useVModel(props, 'modelValue', emits)
+
+let isExpanded = $ref(false)
+
+let isLoading = $ref(false)
+
+const [latitude, longitude] = (vModel.value || '').split(';')
+
+const latLongStr = computed(() => {
+  const [latitude, longitude] = (vModel.value || '').split(';')
+  return latitude && longitude ? `${latitude}; ${longitude}` : 'Set location'
+})
+
+const formState = reactive({
+  latitude,
+  longitude,
+})
+
+const handleFinish = () => {
+  vModel.value = latLongToJoinedString(parseFloat(formState.latitude), parseFloat(formState.longitude))
+  isExpanded = false
+}
+
+const clear = () => {
+  isExpanded = false
+
+  formState.latitude = latitude
+  formState.longitude = longitude
+}
+
+const onClickSetCurrentLocation = () => {
+  isLoading = true
+  const onSuccess = (position) => {
+    const crd = position.coords
+    console.log('crd', crd)
+    formState.latitude = crd.latitude
+    formState.longitude = crd.longitude
+    isLoading = false
+  }
+
+  const onError = (err) => {
+    console.warn(`ERROR(${err.code}): ${err.message}`)
```

Comment:
if it is error, then `console.error` would be better.

Reply:
done